### PR TITLE
tiltfile: allow escaping of selector strings

### LINKED ISSUE #3761: Tilt kubernetes object selector should allow escaping for ':' in resource name
Currently it is not possible to reference a resource using the tilt-specific kubernetes object selector if that resource has a name with a ':' in it.

For example, if using the helm_remote extension to deploy cert-manager, it is not possible to group a subset of the created roles and clusterroles created as part of a k8s_resource.

Example of the created roles/clusterroles that exhibit this behavior:
```
     → cert-manager-cainjector:leaderelection:role
     → cert-manager:leaderelection:role
     → cert-manager-webhook:dynamic-serving:role
     → cert-manager-cainjector:leaderelection:rolebinding
     → cert-manager:leaderelection:rolebinding
     → cert-manager-webhook:dynamic-serving:rolebinding
```

### PR DESCRIPTION
**Pull request by landism**
fixes #3761

I wanted to use some existing library for splitting w/ escapes, but I failed to find one.

### CODE REVIEW

File: internal/sliceutils/sliceutils.go [hunk 1]
```diff
@@ -77,3 +77,73 @@ func AppendWithoutDupes(a []string, b ...string) []string {
...
+				snippetStart := i - 3
+				if snippetStart < 0 {
+					snippetStart = 0
+				}
+				snippetEnd := i + 3
+				if snippetEnd > len(s) {
+					snippetEnd = len(s)
+				}
+
+				return nil, fmt.Errorf("invalid escape sequence '%c%c' in '%s'", opts.EscapeChar, r, s[snippetStart:snippetEnd])
```
Comment (nicks) [inline internal/sliceutils/sliceutils.go:112]: question: this syntax already allows regexps between the colons right? doesn't that mean you can't escape things for the regexp anymore?

**Review by nicks — APPROVED** [review]
I'm fine with moving forward on this in the short-term, but I have long-term concerns that this doubles down on an bespoke object selector syntax that no one else uses and doesn't give us any kind of special advantage. 

I wish we had just used a thin starlark layer over ObjectReference (https://pkg.go.dev/k8s.io/api/core/v1?tab=doc#ObjectReference), like objects=[objectReference(name='foo')]

or even better, LabelSelector (https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1?tab=doc#LabelSelector)

File: internal/sliceutils/sliceutils.go (hunk 1 above)
Comment (landism) [reply internal/sliceutils/sliceutils.go:112]: Nope - the `ObjectSelector` type allows regexes, but the function that converts this syntax into an `ObjectSelector` [calls `regexp.QuoteMeta`](https://github.com/tilt-dev/tilt/blob/816e9a6a9d150f8b543c7bb2112a2c558caff0cd/internal/k8s/object_selector.go#L100) on all of its components.

Other places that use the `ObjectSelector` type and do allow regexes (also highlighting the need for cleaning this up): `k8s_kind`, `k8s_image_json_path`, `filter_yaml`

**Comment by landism** [comment]
> I have long-term concerns that this doubles down on an bespoke object selector syntax that no one else uses and doesn't give us any kind of special advantage.

Yeah, this seems like a good conversation to have at some point.

**Pull request merged**